.commitlintrc.yml

```diff
@@ -0,0 +1,62 @@
+# Commitlint configuration.
+# See: https://github.com/conventional-changelog/commitlint/blob/master/docs/reference-rules.md
+---
+rules:
+  body-leading-blank:
+    - 2
+    - always
+  body-max-line-length:
+    - 2
+    - always
+    - 75
+  footer-leading-blank:
+    - 2
+    - always
+  header-max-length:
+    - 2
+    - always
+    - 75
+  scope-case:
+    - 2
+    - never
+    - - sentence-case
+      - start-case
+  subject-case:
+    - 2
+    - always
+    - - pascal-case
+      - sentence-case
+      - start-case
+      - upper-case
+  subject-empty:
+    - 2
+    - never
+  subject-full-stop:
+    - 2
+    - never
+    - .
+  type-case:
+    - 2
+    - always
+    - lower-case
+  type-empty:
+    - 2
+    - never
+  type-enum:
+    - 2
+    - always
+    - - build
+      - chore
+      - ci
+      - deps
+      - docs
+      - feat
+      - fix
+      - perf
+      - refactor
+      - revert
+      - style
+      - test
```

Comment:
I'm note sure about the "test" type. When I correct styling in a test, would I choose "test" or "style"? Also, the type should be, well, about the type of the change, not about the area of code it touches. And "test" sounds to me more like an area than a type.

Reply:
From https://github.com/angular/angular/blob/main/CONTRIBUTING.md#-commit-message-format:

```
test: Adding missing tests or correcting existing tests
```